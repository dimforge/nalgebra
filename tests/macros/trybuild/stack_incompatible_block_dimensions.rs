use nalgebra::{matrix, stack};

fn main() {
    // Use multi-letter names for checking that the reported span comes out correctly
    let a11 = matrix![1, 2;
                      3, 4];
    let a12 = matrix![5, 6;
                      7, 8];
    let a21 = matrix![9, 10, 11];
    let a22 = matrix![12, 13];
    stack![a11, a12;
           a21, a22];
}