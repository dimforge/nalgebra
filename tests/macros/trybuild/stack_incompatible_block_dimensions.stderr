error[E0277]: the trait bound `ShapeConstraint: SameNumberOfColumns<Const<2>, Const<3>>` is not satisfied
  --> tests/macros/trybuild/stack_incompatible_block_dimensions.rs:12:12
   |
12 |            a21, a22];
   |            ^^^ the trait `SameNumberOfColumns<Const<2>, Const<3>>` is not implemented for `ShapeConstraint`
   |
   = help: the following other types implement trait `SameNumberOfColumns<D1, D2>`:
             <ShapeConstraint as SameNumberOfColumns<D, D>>
             <ShapeConstraint as SameNumberOfColumns<D, Dyn>>
             <ShapeConstraint as SameNumberOfColumns<Dyn, D>>
   = note: this error originates in the macro `stack` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0282]: type annotations needed
  --> tests/macros/trybuild/stack_incompatible_block_dimensions.rs:11:5
   |
11 | /     stack![a11, a12;
12 | |            a21, a22];
   | |____________________^ cannot infer type
   |
   = note: this error originates in the macro `stack` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0599]: no method named `generic_view_mut` found for struct `Matrix<_, Const<3>, _, _>` in the current scope
  --> tests/macros/trybuild/stack_incompatible_block_dimensions.rs:11:5
   |
11 |       stack![a11, a12;
   |  _____^
12 | |            a21, a22];
   | |____________________^ method not found in `Matrix<_, Const<3>, _, _>`
   |
  ::: src/base/matrix_view.rs
   |
   |        generic_slice_mut => generic_view_mut,
   |                             ---------------- the method is available for `Matrix<_, Const<3>, _, _>` here
   |
   = note: the method was found for
           - `Matrix<T, R, C, S>`
   = note: this error originates in the macro `stack` (in Nightly builds, run with -Z macro-backtrace for more info)
